ts/Core/Series/Series.ts

```diff
@@ -1327,9 +1327,26 @@ class Series {
             updatedData,
             indexOfX = 0,
             indexOfY = 1,
-            firstPoint = null;
+            firstPoint = null,
+            copiedData;
+
+        // If this is the first time we're writing back to options, and
+        // allowMutatingData enabled createa copy so that we don't mutate the
+        // source array. (#4259)
```

Comment:
Typo "createa"

Sorry but I don't understand some sentences here, e.g.
```we're writing back to options```

or
```and allowMutatingData enabled createa (should it be "created" or "creates" or "creates a"?) copy so that we don't mutate the source array```